pex/scie/science.py

```diff
@@ -0,0 +1,320 @@
+                        PEX_INTERPRETER="1",
+                        _PEX_SCIE_INSTALLED_PEX_DIR=installed_pex_dir,
+                        # We can get a warning about too-long script shebangs, but this is not
+                        # relevant since we above run the PEX via python and not via shebang.
+                        PEX_EMIT_WARNINGS="0",
+                    ),
+                ),
+                "name": "configure",
+                "exe": "#{cpython:python}",
+                "args": ["{pex}", "{configure-binding.py}"],
+            }
+        ],
+    }  # type: Dict[str, Any]
+
+    for target in configuration.targets:
+        manifest_path = os.path.join(
+            safe_mkdtemp(),
+            target.platform.qualified_file_name("{name}-lift.toml".format(name=name)),
+        )
+        with safe_open(manifest_path, "w") as fp:
+            toml.dump(
+                {
+                    "lift": dict(
+                        lift,
+                        platforms=[target.platform.value],
+                        interpreters=[
+                            {
+                                "id": "cpython",
+                                "provider": "PythonBuildStandalone",
+                                "release": target.pbs_release,
+                                "version": target.version_str,
+                                "lazy": configuration.options.style is ScieStyle.LAZY,
+                            }
+                        ],
+                    )
+                },
+                fp,
+            )
+        yield Manifest(target=target, path=manifest_path)
+
+
+def _science_dir(
+    env,  # type: Variables
+    *components  # type: str
+):
+    # type: (...) -> str
+    return os.path.join(env.PEX_ROOT, "scies", "science", str(MIN_SCIENCE_VERSION), *components)
+
+
+def _qualified_science_binary_name():
+    # type: () -> str
+    return SciePlatform.current().qualified_binary_name("science")
+
+
+def _science_binary_names():
+    # type: () -> Iterator[str]
+    yield "science"
+    yield _qualified_science_binary_name()
+
+
+def _path_science():
+    # type: () -> Optional[str]
+    for path_element in os.environ.get("PATH", os.defpath).split(os.pathsep):
+        for binary in (
+            os.path.join(path_element, binary_name) for binary_name in _science_binary_names()
+        ):
+            if not is_exe(binary):
+                continue
+            try:
+                if (
+                    Version(subprocess.check_output(args=[binary, "--version"]).decode("utf-8"))
+                    < MIN_SCIENCE_VERSION
+                ):
+                    continue
+            except (CalledProcessError, InvalidVersion):
+                continue
+            return binary
+    return None
+
+
+def _science_binary_url(suffix=""):
+    # type: (str) -> str
+    return "https://github.com/a-scie/science/releases/download/v{version}/{binary}{suffix}".format(
```

Comment:
Didn't these used to be located at `a-scie/lift`? Has that/Will that change?

Reply:
Facepalm. Thanks! I think this explains the CI failures which are 404 and not 403 - which is I think how GH indicates a rate limit.